Sources/StoryboardInstantiatable.swift

```diff
@@ -73,6 +73,36 @@ extension DependencyContainer {
   public func resolveDependencies<T>(of instance: T, tag: Tag? = nil) throws {
     _ = try resolve(tag: tag) { (_: () throws -> T) in instance }
   }
+  
+  /**
+   Register factory for type `T` which has to conform to `StoryboardInstantiatable` and associate it with an optional tag.
+   
+   - parameters:
+   - scope: The scope to use for instance created by the factory. Default value is `Shared`.
+   - type: Type to register definition for. Default value is return value of factory.
+   - tag: The arbitrary tag to associate this factory with. Pass `nil` to associate with any tag. Default value is `nil`.
+   - factory: The factory that produces instance of `type`. Will be used to resolve instances of `type`.
+   
+   - returns: A registered definition.
+   
+   - note: You should cast the factory return type to the protocol you want to register it for
+   (unless you want to register concrete type) or provide `type` parameter.
+   
+   - seealso: `Definition`, `ComponentScope`, `DependencyTagConvertible`
+   
+   **Example**:
+   ```swift
+   //Register MyViewController
+   container.register(tag: "myVC") { MyViewController() }
+   .resolvingProperties { container, controller in
+      controller.service = try container.resolve() as MyViewControllerService
+   }
+   
+   ```
+   */
+  public func registerViewController<T: StoryboardInstantiatable>(_ scope: ComponentScope = .shared, type: T.Type = T.self, tag: DependencyTagConvertible? = nil, factory: @escaping () throws -> T) -> Dip.Definition<T, ()> {
```

Comment:
I'm not sure it will work even if we introduce `as` parameter, as there is no way to constraint generic parameters in such way: `func register<T: NSObject, AsType>(storyboardType type: T.Type, as: AsType...) where T: StoryboardInstantiatable, T: AsType`
So I would say it's either that or a regular `register` that does not check for StoryboardInstantiatable. And there still will be a way to provide protocol type via type forwarding